Modules/Module_Processing/Module_ASL/xASL_wrp_ResampleASL.m

```diff
@@ -254,84 +253,90 @@ function xASL_wrp_ResampleASL(x)
-        fprintf('%s\n', PathPWI{iSpace});
-        xASL_io_SaveNifti(PathASL{iSpace}, PathPWI{iSpace}, PWIsingle, 32, false);
         
+    % =====================================================================
+    % C) Single- and multi-PLD subtraction
     else
-        % Paired subtraction
-        [ControlIm, LabelIm] = xASL_quant_GetControlLabelOrder(ASL_im);
-        ASL_im = ControlIm - LabelIm;
-
-        % Average PWI - multiPLD
-        if x.modules.asl.bMultiPLD 
-			% Skip every other value in x.Q.Initial_PLD as it was stored for both control and label images 
-			% and we need the PLD vector now for the pairwise subtractions only
-			Initial_PLD_PWI = x.Q.Initial_PLD(1:2:end);
-
-			% The labeling durations also need to be taken into account and unique combinations of PLDs with LDs should be considered
-			if numel(x.Q.Initial_PLD) == numel(x.Q.LabelingDuration)
-				Initial_LabDur_PWI = x.Q.LabelingDuration(1:2:end);
-			else
-				Initial_LabDur_PWI = ones(size(Initial_PLD_PWI)) * x.Q.LabelingDuration;
-			end
+        % Firstly, equalize PLD for single- and multi-PLD
+        nPLD = length(x.Q.Initial_PLD);
+        factorPLD = nVolumes/nPLD;
+        initialPLD = repmat(x.Q.Initial_PLD, [factorPLD 1]);
+        % this vector now has the length of the number of volumes
+        % either all values are identical (in the case of single-PLD)
+        % or a combination of multi-PLD
+        
+        % Secondly, equalize LabDur for single- and multi-PLD
+        nLabDur = length(x.Q.LabelingDuration);
+        factorLabDur = nVolumes/nLabDur;
+        LabDuration = repmat(x.Q.LabelingDuration, [factorLabDur 1]);
         
+        if dim4>1 && ~x.modules.asl.bContainsDeltaM
+            % Paired subtraction
+            [ControlIm, LabelIm] = xASL_quant_GetControlLabelOrder(ASL_im);
+            PWI4D = ControlIm - LabelIm;
+            
+            % Skip every other value in x.Q.Initial_PLD as it was stored for both control and label images 
+            % we need the PLD vector now for the pairwise subtractions only            
+            initialPLD_PWI4D = initialPLD(1:2:end);
+            % Do the same for LabDur
+            LabDuration_PWI4D = LabDuration(1:2:end);
+
 			% After averaging across PLDs, we'll obtain these unique PLDs+LD combinations
-			% indexNew lists for each original position to where it should be averaged
-			[~, ~, indexNew] = unique([Initial_PLD_PWI, Initial_LabDur_PWI], 'stable', 'rows');
-			
-            % MultiPLD PWI after averaging
-			PWI = zeros(size(ASL_im,1), size(ASL_im,2), size(ASL_im,3), max(indexNew)); % preallocate PWI
-            for nPLD = 1:max(indexNew)
-                PWI(:, :, :, nPLD) = xASL_stat_MeanNan(ASL_im(:, :, :, indexNew == nPLD), 4); % Averaged PWI4D 
+			% indexAverage_PLD_LabDur lists for each original position to where it should be averaged
+			[~, ~, iUnique_PLD_LabDur] = unique(unique([initialPLD_PWI4D, LabDuration_PWI4D], 'stable', 'rows'), 'stable', 'rows');

```

Comment:
Why do you call `unique` twice?

Reply:
Cause it's wrong to call it twice and it won't do what it should!

Reply:
Because there were 2 dims to unique from, I tested this and it worked in a multi-PLD dataset. Using unique multiple times never hurts if you already only have unique elements.

Reply:
Yes, it works in the sense of not crashing. But it's not doing what it should! 
The whole point of this function is to create a vector of the same length as the original with indices for groups of unique PLD/LabDur combinations.

So for PLDs [100 100 100 200 200 200 300] and LabDur [100 100 200 200 200 200 200], it will do [1 1 2 3 3 4] - so it will nicely group all the nice combinations of PLD/LabDur and create a vector of the original length that can be used for indexing PWI4D.

If you call unique twice, you end up with [1 2 3 4] - that's a list of indices that is useless as it cannot be used for the original purpose.

Please revert back to the original working code. This code won't work.